### PR TITLE
Implement reading PDO from OD defaults (#273)

### PR DESCRIPTION
**Pull request by sveinse**
A small improvement based on discussion in #273 that allows reading the PDO configuration from the object dictionary _defaults_ rather than requesting them from the remote node.

### CODE REVIEW

**Comment by samsamfire** [comment]
Hi, I did something similar on my side for two main reasons (just putting this for explanation):

- When using a large OD, reading via SDO can be quite long : ~5 seconds. When you are testing and just running scripts, this can be really annoying.
- Sometimes you would like to use PDOs but don't want to use SDO as this can break communication if there is already a master on the bus talking to the node.

Anyway, cool improvement !

File: canopen/pdo/base.py [hunk 1]
```diff
@@ -313,43 +313,49 @@ def add_callback(self, callback: Callable[["Map"], None]) -> None:
         """
         self.callbacks.append(callback)
 
-    def read(self) -> None:
+    def read(self, from_od=False) -> None:
         """Read PDO configuration for this map using SDO."""
-        cob_id = self.com_record[1].raw
+
+        def _raw_from(param):
```
Comment (acolomb) [inline canopen/pdo/base.py:319]: Do we need an underscore name here?  The function is only defined within this scope, so there is nothing to be gained by trying to hide it?

**Review by acolomb — COMMENTED** [review]
Nice addition.  Only one nit-pick comment.

**Comment by acolomb** [comment]
And I think this needs documentation?

File: canopen/pdo/base.py (hunk 1 above)
Comment (sveinse) [reply canopen/pdo/base.py:319]: No, not really. Only a convention. My habit is to name functions that are defined within a fn or meth using a underscore prefix, but it is not necessity.

**Pull request merged**